compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/collectors/components/DeclarationCheckersDiagnosticComponent.kt

```diff
@@ -13,65 +14,66 @@ import org.jetbrains.kotlin.fir.analysis.collectors.AbstractDiagnosticCollector
 import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
 import org.jetbrains.kotlin.fir.declarations.*
 
+@OptIn(CheckersComponentInternal::class)
 class DeclarationCheckersDiagnosticComponent(
     collector: AbstractDiagnosticCollector,
     private val checkers: DeclarationCheckers = collector.session.checkersComponent.declarationCheckers,
 ) : AbstractDiagnosticCollectorComponent(collector) {
 
     override fun visitFile(file: FirFile, data: CheckerContext) {
-        checkers.fileCheckers.check(file, data, reporter)
+        checkers.allFileCheckers.check(file, data, reporter)
     }
 
     override fun visitProperty(property: FirProperty, data: CheckerContext) {
-        (checkers.memberDeclarationCheckers + checkers.propertyCheckers).check(property, data, reporter)
+        checkers.allPropertyCheckers.check(property, data, reporter)
     }
 
     override fun <F : FirClass<F>> visitClass(klass: FirClass<F>, data: CheckerContext) {
-        checkers.classCheckers.check(klass, data, reporter)
+        checkers.allClassCheckers.check(klass, data, reporter)
     }
 
     override fun visitRegularClass(regularClass: FirRegularClass, data: CheckerContext) {
-        (checkers.memberDeclarationCheckers + checkers.regularClassCheckers).check(regularClass, data, reporter)
+        checkers.allRegularClassCheckers.check(regularClass, data, reporter)
     }
 
     override fun visitSimpleFunction(simpleFunction: FirSimpleFunction, data: CheckerContext) {
-        (checkers.memberDeclarationCheckers + checkers.functionCheckers).check(simpleFunction, data, reporter)
+        (checkers.memberDeclarationCheckers + checkers.allFunctionCheckers).check(simpleFunction, data, reporter)
```

Comment:
`allFunctionCheckers` should instead include `checkers.memberDeclarationCheckers`

Reply:
As replied above to @jsjeon's comment, to do this we need to limit the scope `FirFunctionChecker` to only check member declarations.

Reply:
So, not all `FirFunction<*>` implements `FirMemberDeclaration`. I see... and that's why this is the only place with an exception: everything else now uses `allXXX`, which looks better. Perhaps leave code comments to explain this?

Reply:
+1 to comments. I missed this too on first read, actually

Reply:
Actually, wouldn't it be better to add `allSimpleFunctionCheckers`? We have `allRegularClassCheckers` vs just `allClassCheckers`. I don't see why we couldn't have for simple function.

Reply:
Adding `FirSimpleFunctionChecker` alias sounds good!

Reply:
Thanks! Adding a simple function nchecker sounds good!